include/DiscoveryServerManager.h

```diff
@@ -187,11 +187,11 @@ class DiscoveryServerManager
     DiscoveryServerManager(
             const std::string& xml_file_path,
             const bool shared_memory_off);
-#if FASTRTPS_VERSION_MAJOR >= 2 && FASTRTPS_VERSION_MINOR >= 2
+#if FASTDDS_VERSION_MAJOR >= 2 && FASTDDS_VERSION_MINOR >= 2
     FASTDDS_DEPRECATED_UNTIL(3, "eprosima::discovery_server::DiscoveryServerManager(const std::set<std::string>& xml_snapshot_files,"
```

Comment:
You would need to remove this because this will throw an assertion if building with Fast DDS 3. You should remember to not only remove the declaration but also the implementation from the source file.